src/metabase/db/data_migrations.clj

```diff
@@ -237,16 +237,16 @@
     (db/simple-delete! Setting {:key "enable-advanced-humanization"})))
 
 ;; Starting in version 0.29.0 we switched the way we decide which Fields should get FieldValues. Prior to 29, Fields
-;; would be marked as special type Category if they should have FieldValues. In 29+, the Category special type no
+;; would be marked as semantic type Category if they should have FieldValues. In 29+, the Category semantic type no
 ;; longer has any meaning as far as the backend is concerned. Instead, we use the new `has_field_values` column to
 ;; keep track of these things. Fields whose value for `has_field_values` is `list` is the equiavalent of the old
-;; meaning of the Category special type.
+;; meaning of the Category semantic type.
 ;;
 ;; Since the meanings of things has changed we'll want to make sure we mark all Category fields as `list` as well so
 ;; their behavior doesn't suddenly change.
 (defmigration ^{:author "camsaul", :added "0.29.0"} mark-category-fields-as-list
   (db/update-where! Field {:has_field_values nil
-                           :special_type     (mdb.u/isa :type/Category)
+                           :semantic_type    (mdb.u/isa :type/Category)
```

Comment:
I don't know how the timing between data and DB migrations work—is it correct for these old data migrations to use the new column name?

Reply:
really good question. the docstring from the ns seems to say it could be both:

>   These migrations are all ran once when Metabase is first launched, except when transferring data from an existing
  H2 database.  When data is transferred from an H2 database, migrations will already have been run against that data;
  thus, all of these migrations need to be repeatable, e.g.:

I'm not sure how to handle that besides catching errors and trying again.

Reply:
@camsaul have you run into this before?